Clean up old / stray Azure resources before the test run, speed up tests

Kami opened the pull request:
This pull request updates Azure Blobs integration test class so we clean any old stray resource groups left from previous test runs before we actually run the tests.

This should ensure we don't have any limits and we delete any stray resources from previous runs (we can end up with stray resources if the CI build is terminated non-gracefully and ``tearDownClass`` has no chance to run).

Kami committed:
Update setUpClass method in the Azure integration tests class to clean
up any stray / left over resources from previous runs.

If CI runs gets killed or aborted non-gracefully, there will be left
over resources left laying around.

To ensure we don't hit any API limits / quotas we now always perform
this clean up step during run.

For initial run, delete all the old groups. Once that run will complete,
I will comment out the code where we also perform timestamp check to
ensure we only delete groups older then a couple of days - this way we
ensure we don't delete any groups from other concurrent test runs.

c-w commented on integration/storage/test_azure_blobs.py:
```diff
@@ -128,12 +129,35 @@ def setUpClass(cls):
+        # couple (6) of hours
+        resource_groups = resource_client.resource_groups.list()
+        now_ts = int(time.time())
+        delete_threshold_ts = now_ts - (6 * 60 * 60)
```
Nit: this may be easier to read using `from datetime import timedelta; timedelta(hours=60).total_seconds()`

c-w commented on integration/storage/test_azure_blobs.py:
```diff
@@ -128,12 +129,35 @@ def setUpClass(cls):
+
+            if resource_group.name.startswith(name) and resource_group.location == location and \
+               'test' in resource_groups.tags and resource_create_ts <= delete_threshold_ts:
+               #'test' in resource_groups.tags and resource_create_ts <= delete_threshold_ts:
```
Nit: remove comment before merge.

c-w commented on integration/storage/test_azure_blobs.py:
```diff
@@ -128,12 +129,35 @@ def setUpClass(cls):
+                try:
+                    resource_client.resource_groups.begin_delete(resource_group.name)
+                except Exception as e:
+                    print("Failed to delete resource group: %s" % (str(e)))
```
Nit: suggest to log to stderr so it gets highlighted in the logs: `from sys import stderr; print("Failed to delete resource group: %s" % (str(e)), file=stderr)`

Kami committed:
Test the change.

Kami committed:
Make sure we pass GITHUB_ env variables to the tox run.

Kami committed:
Update code per PR feedback, test the change again.

Kami committed:
Use latest version of pytest for integration tests, use tee-sys
functionality so produced output is printed to console in real time and
also on success and not just on failure.

Kami committed:
Fix bug in the clean up code.

Kami committed:
Add additional print and assert.

Kami committed:
Remove testing changes.

Kami committed:
Include additional tags.

Kami commented:
@c-w Thanks for the review.

I pushed some more changes and fixes and I believe it should be working now. I also reverted the testing change to allow for concurrent runs.

Kami committed:
Use try / finally.

c-w commented on integration/storage/test_azure_blobs.py:
```diff
@@ -124,17 +128,46 @@ def setUpClass(cls):
+            if resource_group.name.startswith(RESOURCE_GROUP_NAME_PREFIX) and \
+               resource_group.location.lower() == location.lower() and \
+               'test' in resource_group.tags and resource_create_ts <= delete_threshold_ts:
+                assert resource_group.name.startswith(RESOURCE_GROUP_NAME_PREFIX)
```
Is this necessary given that we have the check on line 149 which means that this assert should always pass?

Kami replied:
It's jut an additional safe guard in case there was a fat finger in if or similar which could potentially result in objects with a different prefix to be deleted (granted, it's probably an over kill since we have additional checks and right now that account is only used for those tests).

Kami committed:
Use larger chunk size to speed up the tests.

Previously we used chunk size of 1 which resulted in 1 * 1024 * 1024
iterations and API request which was slow and meant that test took ~70
seconds.

Kami committed:
Make sure value is an int, use valid prefix.

Kami committed:
Update more tests to use less chunks to speed up all tests which
previously used 1 byte chunk size.

Kami commented:
I noticed some test cases took almost 100 seconds, so I pushed a change which should speed it up - d31e3ee65768dbaa05bf85f85feb90c20f64749b, bbcaa2748fa96c607b6ac8252aff415ac30facbe.

This will substantially reduce number of iterations (previously for some test cases we used 1 * 1024 * 1024 iterations which will be slow), but there will still be multiple ones so that iteration / chunk upload functionality will still be tested.

Kami commented:
https://github.com/apache/libcloud/runs/3980290729?check_suite_focus=true

Looks like the integration tests run with that change now takes ~1 minute instead of ~8 minutes.

Kami committed:
Add additional assert.

Kami committed:
Reduce workflow timeout now that we have substantially sped up the
tests.

Kami committed:
Pass -rsx flag to pytest to report reason for skipped tests.
